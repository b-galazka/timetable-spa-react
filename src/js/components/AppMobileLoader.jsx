import React from 'react';
import { connect } from 'react-redux';
import propTypes from 'prop-types';

import LoadingAnimation from './LoadingAnimation';
import errors from '../../json/errors';

import '../../scss/mobileAppLoader.scss';

function mapStateToProps(state) {

    const { fetchingError } = state.mobileAppData;

    return {
        mobileAppDataFetchingError: fetchingError
    };
}

function AppMobileLoader({ mobileAppDataFetchingError }) {

    return (
        <section className="mobile-app__loader">

            {
                (mobileAppDataFetchingError) ?

                    <p className="mobile-app__fetching-error">{errors.fetchingError}</p> :

                    <figure>
                        <LoadingAnimation width="120px" height="120px" />
                    </figure>
            }

        </section>
    );
}

AppMobileLoader.propTypes = {
    // redux
    mobileAppDataFetchingError: propTypes.bool.isRequired
};

export default connect(mapStateToProps)(AppMobileLoader);