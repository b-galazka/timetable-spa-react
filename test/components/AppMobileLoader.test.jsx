import React from 'react';
import renderer from 'react-test-renderer';

import AppMobileLoader from '../../src/js/components/AppMobileLoader';

jest.mock('../../src/js/components/LoadingAnimation', () => 'LoadingAnimation');
jest.mock('../../src/json/errors', () => ({ fetchingError: 'fetching error' }));

describe('AppMobileLoader component', () => { 

    it('should render loading animation if there is no fetching error', () => {

        const tree = renderer.create(
            <AppMobileLoader.WrappedComponent
                mobileAppDataFetchingError={false}
            />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should render fetching error', () => {

        const tree = renderer.create(
            <AppMobileLoader.WrappedComponent
                mobileAppDataFetchingError
            />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});