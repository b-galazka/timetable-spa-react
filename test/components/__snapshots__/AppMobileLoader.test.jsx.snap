// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppMobileLoader component should render fetching error 1`] = `
<section
  className="mobile-app__loader"
>
  <p
    className="mobile-app__fetching-error"
  >
    fetching error
  </p>
</section>
`;

exports[`AppMobileLoader component should render loading animation if there is no fetching error 1`] = `
<section
  className="mobile-app__loader"
>
  <figure>
    <LoadingAnimation
      height="120px"
      width="120px"
    />
  </figure>
</section>
`;
